### ISSUE TITLE
Service interface not documented in howto

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5586]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5586|
|Type|defect|
|Created|2012-03-30 23:25:19Z|
|Branch|https://github.com/twisted/twisted/tree/service-docs-5586|

Creating new services is not documented anywhere in our howtos, except one mention in the tutorials.

<details><summary>Searchable metadata</summary>

```
trac-id__5586 5586
type__defect defect
reporter__itamar itamar
priority__high high
milestone__ 
branch__branches_service_docs_5586 branches/service-docs-5586
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1333149919000000 1333149919000000
changetime__1368396271000000 1368396271000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst set owner to @itamarst|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

(In [37514]) Branching to 'service-docs-5586'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst removed owner|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/service-docs-5586

Ready for review, I hope.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_4"></a>|@wsanchez commented|
|-|-|

Minor nit: r37619

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_5"></a>|@wsanchez set owner to @itamarst|
|-|-|

Reviewed.

This is a simple and nice improvement on the application howto.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_6"></a>|@wsanchez commented|
|-|-|

There was a "+1" in the last comment, until somehow there was not.  Anyway: +1

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

Looks like this is ready to be merged?

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_8"></a>|@itamarst commented|
|-|-|

Yes, just been too busy/tired to do it. Feel free.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [38397]) Merge service-docs-5586: The Application howto now provides an example of writing a custom Service.

Author: itamarst
Reviewer: wsanchez
Fixes: [#5586](https://github.com/twisted/twisted/issues/5586)
